### PR TITLE
Organizing osquery python testing

### PR DESCRIPTION
**Pull request by theopolis**
Move /osquery/python_tests/* to /tools/tests
Move test_extensions process controls to test_base module
Use test_base.Testing to implement each module's main()
  - This applies a default argparse with --build
  - test_base.ARGS is the argparse-parsed namespace
  - Use test_base.ARGS.build for the platform-specific dir
Move WatchdogTests to /tools/tests/test_watchdog.py

### CODE REVIEW

**Comment by theopolis** [comment]
@zwass, what do you think of this pattern? 

Put all python-tests into `/tools/tests/` and expect `test_base.ARGS.build` to be set with CMake's binary output directory.

```python
if __name__ == '__main__':
  test_base.Testing.run()
```

Will apply the `argparse.parse_args()` and if you need to act before dropping into `unittest.main()`, like extensions does:

```python
if __name__ == "__main__":
    module = test_base.Tester()

    # Find and import the thrift-generated python interface
    thrift_path = test_base.ARGS.build + "/generated/gen-py"
    try:
        sys.path.append(thrift_path)
        from osquery import *
    except ImportError:
        print ("Cannot import osquery thrift API from %s" % (thrift_path))
        print ("You must first run: make")
        exit(1)

    module.run()
```

File: Makefile [hunk 1]
```diff
@@ -16,41 +16,43 @@ else
 	endif
 endif
 
+DEFINES := CTEST_OUTPUT_ON_FAILURE=1
```
Comment (zwass) [inline Makefile]: Nice. This should be handy... I've been using the logs it stores in `build/darwin/Testing/Temporary/LastTest.log`.

File: tools/tests/test_base.py [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+        )
+        parser.add_argument(
+            "--verbose", default=False, action="store_true",
+            help="Run daemons and extensions with --verbose"
+        )
+
+        # Directory structure options
+        parser.add_argument(
+            "--build", metavar="PATH", default=".",
+            help="Path to osquery build (./build/<sys>/)."
```
Comment (zwass) [inline tools/tests/test_base.py]: Pulling in the build dir like this is a really good idea.

File: CMake/CMakeLibs.cmake [hunk 3]
```diff
@@ -28,8 +28,10 @@ macro(ADD_OSQUERY_TEST IS_CORE TEST_NAME SOURCE)
 endmacro(ADD_OSQUERY_TEST)
 
 macro(ADD_OSQUERY_PYTHON_TEST TEST_NAME SOURCE)
-  SYMLINK_FILE(${CMAKE_CURRENT_LIST_DIR}/${SOURCE} ${CMAKE_BINARY_DIR}/osquery/python_tests/${SOURCE})
-  add_test(NAME python_${TEST_NAME} COMMAND python ${CMAKE_BINARY_DIR}/osquery/python_tests/${SOURCE})
+  #SYMLINK_FILE(${CMAKE_CURRENT_LIST_DIR}/${SOURCE} ${CMAKE_BINARY_DIR}/osquery/python_tests/${SOURCE})
```
Comment (zwass) [inline CMake/CMakeLibs.cmake]: Might as well remove this (and the macro for it) if we're going to run the tests out of the source dir.

**Comment by zwass** [comment]
Really glad to see that we'll be able to run these Python based integration tests from the regular CTest execution. I think having more integration tests and having them run on each diff is going to be super helpful in defining expected behaviors and preventing regressions.

**Pull request merged**